src/engine/controls/cuecontrol.cpp

```diff
@@ -1715,20 +1746,23 @@ void CueControl::hotcueFocusColorNext(double v) {
         return;
     }
 
-    PredefinedColorPointer pColor = pControl->getColor();
-    if (!pColor) {
+    mixxx::RgbColor::optional_t controlColor = pControl->getColor();
+    if (!controlColor) {
         return;
     }
+    mixxx::RgbColor color = *controlColor;
+
+    ColorPaletteSettings colorPaletteSettings(m_pConfig);
+    ColorPalette colorPalette = colorPaletteSettings.getHotcueColorPalette();
 
-    // Get next color in color set
-    int iColorIndex = Color::kPredefinedColorsSet.predefinedColorIndex(pColor) + 1;
-    if (iColorIndex >= Color::kPredefinedColorsSet.allColors.size()) {
+    // Get previous color in color set
```

Comment:
\+ 1 = previous?

Why do we do this prev/next calculation based on indexOf() here instead of adding dedicated functions for these tasks to ColorPalette? Cycling through the available colors is one of its purposes.